semgrep-core/src/engine/Eval_generic.ml

```diff
@@ -155,13 +155,33 @@ let rec eval env code =
   (* Python int() operator *)
   | G.Call ({ e = G.N (G.Id (("int", _), _)); _ }, (_, [ Arg e ], _)) -> (
       let v = eval env e in
+      logger#trace "called int(%s)" (show_value v);
       match v with
       | Int _ -> v
       | String s -> (
           match int_of_string_opt s with
           | None -> raise (NotHandled code)
           | Some i -> Int i)
       | __else__ -> raise (NotHandled code))
+  (* Custom digits() operator
+   * This is a virtual builtin helper function provided by Semgrep.
+   * It returns the first group of digits in a value.
+   * It also keeps underscores, and leading minus signs.
+   * The returned value will always cleanly pass through Python's int().
+   * This is useful e.g. to write rules that work with generic captures in CSS.
+   *)
+  | G.Call ({ e = G.N (G.Id (("digits", _), _)); _ }, (_, [ Arg e ], _)) -> (
```

Comment:
Nice! If you don't want to have an `r2c` module then I would at least prefix it with `r2c_` to avoid (future) potential clashes with Python's stdlib.